File: 10 PrimeandComposite/CountFactors/countFactors.java
class Solution {
  public int solution(int N) {
    long i = 1;
    long result = 0;
    while (i * i <= N) {
      if (N % i == 0) {
        if (N / i == i) {
          result++;
        } else {
          result += 2;
        }
      }
      i++;
    }
    return (int) result;
  }
}

File: 10 PrimeandComposite/CountFactors/countFactors.js
function solution(N) {
  let i = 0;
  let result = 0;

  while (i * i <= N) {
    if (N % i === 0) {
      if (i === N / i) {
        result++;
      } else {
        result += 2;
      }
    }
    i++;
  }
  return result;
}

console.log(solution(24));
